core/bin/zksync_server/src/node_builder.rs

```diff
@@ -292,7 +292,11 @@ impl MainNodeBuilder {
         let circuit_breaker_config = try_load_config!(self.configs.circuit_breaker_config);
         let with_debug_namespace = state_keeper_config.save_call_traces;
 
-        let mut namespaces = Namespace::DEFAULT.to_vec();
+        let mut namespaces = if let Some(namespaces) = &rpc_config.api_namespaces {
+            serde_json::from_str(&namespaces.join(","))?
```

Comment:
Will this work? If you want to parse data from a JSON array, you need to wrap an array in `[]` and quote strings (I think `impl Debug for str` does this, but not 100% sure it's always JSON-compatible – it probably is at least for ASCII strings). This looks quite awkward in any case; isn't it possible to e.g. move `Namespace` to `zksync_config` and use it in the config declaration directly?

Reply:
Yes it works.
I don't think we have to move namespaces to the config folder.

It doesn't belong there

Reply:
> Yes it works.

I've run it locally, and it really doesn't.

```text
$ API_WEB3_JSON_RPC_API_NAMESPACES=eth,web3,net,pubsub,zks,en,debug \
  zk server --components api --use-node-framework
2024-06-18T13:53:06.216021Z  INFO zksync_server: No sentry URL was provided
2024-06-18T13:53:06.263693Z  INFO zksync_db_connection::connection_pool: Created DB pool with parameters ConnectionPoolBuilder { database_url: "postgres://***:***@localhost/zksync_local", max_size: 1, acquire_timeout: 30s, statement_timeout: None, db: "zksync_dal::Core" }
Error: expected value at line 1 column 1

Stack backtrace:
   0: std::backtrace::Backtrace::create
   1: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
   2: zksync_server::node_builder::MainNodeBuilder::add_ws_web3_api_layer
   3: zksync_server::node_builder::MainNodeBuilder::build
   4: std::sys_common::backtrace::__rust_begin_short_backtrace
   5: core::ops::function::FnOnce::call_once{{vtable.shim}}
   6: std::sys::pal::unix::thread::Thread::new::thread_start
   7: __pthread_joiner_wake
Error: Child process exited with code 1
```

(These are the default namespaces from [EN config](https://github.com/matter-labs/zksync-era/blob/26f2010ea2edd1cb79d80852c626051afc473c48/etc/env/configs/ext-node.toml#L27).) What am I doing wrong? Am I supposed to ~SQL~ JSON-inject array brackets and quotes into names like this: `API_WEB3_JSON_RPC_API_NAMESPACES='["eth","zks"]'` (that's namespaces `["eth"` and `"zks"]`)? This looks weird and semantically invalid.

Reply:
Decided to switch to strum. Seems it's much better approach